File: BKJ/영단어 암기는 괴로워/pable91.py
# 영단어 암기는 어려워
# https://www.acmicpc.net/problem/20920

a , b = map(int, input().split())

dic = {}

for _ in range(a):
  str = input()

  if len(str) < b:
    continue

  if str in dic:
    dic[str] += 1
  else:
    dic[str] = 1

# 이게 뭔 소리지
sorted_dic = sorted(dic.items(), key=lambda x: (-x[1], -len(x[0]), x[0]))

for key, value in sorted_dic:
  print(key)

File: BKJ/한 줄로 서기/pable91.kt
/**
 *  한 줄로 서기
 *  https://www.acmicpc.net/problem/1138
 *
 */

fun main() {

    val N = readln().toInt()
    val inputs = readln().split(" ").map { it.toInt() }

    val arr = IntArray(N)
    for (i in 0 until inputs.size) {
        val n = inputs[i]
        var cnt = 0
        for (j in 0 until inputs.size) {
            if(cnt == n && arr[j] == 0) {
                arr[j] = i + 1
                break
            }
            if(arr[j] == 0 || arr[j] > i+1) {
                cnt += 1
            }
        }
    }

    for (elem in arr) {
        print("$elem ")
    }

}